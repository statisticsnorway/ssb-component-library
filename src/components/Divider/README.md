Divider
========

> A way of separating content

### Usage

#### HTML

```html
<hr class="ssb-divider type-dark" />
<hr class="ssb-divider type-light" />
```

#### React

```jsx harmony
<Divider light />
<Divider dark />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| className | string | Custom className |
| dark | bool | For dark color |
| light | bool | For light color |
