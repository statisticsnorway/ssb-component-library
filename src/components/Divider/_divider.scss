.ssb-divider {
  border-top: 1px solid;
  display: block;
  height: 2px;
  margin: 0;
  position: relative;
  width: 100%;

  &.type-dark {
    border-top-color: $ssb-dark-5;
  }

  &.type-light {
    border-top-color: $ssb-dark-2;
  }
}
