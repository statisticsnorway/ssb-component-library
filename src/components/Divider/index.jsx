import React from 'react';
import PropTypes from 'prop-types';

const Divider = ({ className, dark, light }) => (
	<hr
		className={`ssb-divider ${className} ${dark ? 'type-dark' : ''} ${light ? 'type-light' : ''}`}
	/>
);

Divider.defaultProps = {
	className: '',
	dark: false,
	light: false,
};

Divider.propTypes = {
	className: PropTypes.string,
	dark: PropTypes.bool,
	light: PropTypes.bool,
};

export default Divider;
