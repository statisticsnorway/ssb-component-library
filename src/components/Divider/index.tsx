import React from 'react'
import classNames from '../../utils/utils'

interface DividerProps {
  className?: string
  dark?: boolean
  light?: boolean
}

const Divider: React.FC<DividerProps> = ({ className, dark = false, light = false }) => (
  <hr className={classNames('ssb-divider', className, dark && 'type-dark', light && 'type-light')} aria-hidden='true' />
)

export default Divider
