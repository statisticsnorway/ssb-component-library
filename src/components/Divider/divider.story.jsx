import React from 'react'
import Divider from './index'

export default {
  title: 'Divider',

  parameters: {
    layout: 'fullscreen',
  },
}

export const Dark = () => (
  <div
    style={{
      alignItems: 'center',
      display: 'flex',
      height: '100%',
      position: 'absolute',
      width: '100%',
    }}
  >
    <Divider dark />
  </div>
)

export const Light = () => (
  <div
    style={{
      alignItems: 'center',
      display: 'flex',
      height: '100%',
      position: 'absolute',
      width: '100%',
    }}
  >
    <Divider light />
  </div>
)
