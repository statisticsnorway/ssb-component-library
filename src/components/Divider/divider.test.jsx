import React from 'react';
import { shallow } from 'enzyme';
import Divider from './index';

describe('Divider component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Divider />);
		expect(wrapper).toMatchSnapshot();
	});
	test('set Light color', () => {
		const wrapper = shallow(<Divider light />);
		expect(wrapper.hasClass('type-light')).toEqual(true);
	});
	test('set Dark color and optional className', () => {
		const wrapper = shallow(<Divider dark className="mt-3"/>);
		expect(wrapper.hasClass('type-dark')).toEqual(true);
		expect(wrapper.hasClass('mt-3')).toEqual(true);
	});
});
