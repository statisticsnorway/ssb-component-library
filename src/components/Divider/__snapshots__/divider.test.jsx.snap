// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component Matches the snapshot 1`] = `
<DocumentFragment>
  <hr
    aria-hidden="true"
    class="ssb-divider"
  />
</DocumentFragment>
`;

exports[`Divider component set Dark color and optional className 1`] = `
<DocumentFragment>
  <hr
    aria-hidden="true"
    class="ssb-divider mt-3 type-dark"
  />
</DocumentFragment>
`;

exports[`Divider component set Light color 1`] = `
<DocumentFragment>
  <hr
    aria-hidden="true"
    class="ssb-divider type-light"
  />
</DocumentFragment>
`;
